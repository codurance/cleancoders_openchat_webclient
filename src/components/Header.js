import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import userService from 'services/User'

class Header extends Component {
  static contextTypes = {
    router: PropTypes.object.isRequired
  }

  logout () {
    userService.logout()
    this.context.router.history.push('/')
  }

  render () {
    const button = (
      <button
        className='nav-item logout btn btn-outline-light'
        onClick={this.logout.bind(this)}
      >
        <i className='fa fa-sign-out' /> Log Out
      </button>)

    const usersToFollow = <Link className='nav-item nav-link mr-3' to='/findUsers'>Follow users</Link>

    return (
      <nav className='navbar navbar-expand navbar-dark bg-primary fixed-top'>
        <Link to='/' className='navbar-brand'>OpenChat</Link>

        <div className='navbar-nav ml-auto'>
          {userService.user ? usersToFollow : null}
          {userService.user ? button : null}
        </div>
      </nav>
    )
  }
}

export default Header
