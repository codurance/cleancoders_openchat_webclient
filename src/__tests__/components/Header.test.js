import React from 'react'
import { shallow } from 'enzyme'
import Header from 'components/Header'

describe('Header', () => {
  it('renders', () => {
    const wrapper = shallow(<Header />)

    expect(wrapper.html()).toMatchSnapshot()
  })
})
