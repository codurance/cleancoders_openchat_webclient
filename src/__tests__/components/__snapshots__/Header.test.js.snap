// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders 1`] = `

<nav class="navbar navbar-expand-md navbar-dark fixed-top bg-dark">
  <a class="navbar-brand"
     href="/"
  >
    OpenChat
  </a>
</nav>

`;
